compiler/rustc_lint/src/unused.rs

```diff
@@ -251,7 +251,7 @@ impl<'tcx> LateLintPass<'tcx> for UnusedResults {
                 }
                 ty::Array(ty, len) => match len.try_eval_usize(cx.tcx, cx.param_env) {
                     // If the array is definitely non-empty, we can do `#[must_use]` checking.
-                    Some(n) if n != 0 => {
+                    Some(n @ 1..) => {
```

Comment:
This is quite cute, but I'm not sure it's more readable. I'd like to hear what others think.

Reply:
My overall hypothesis was that this pattern is neither significantly more or less legible to humans, but significantly more legible to the compiler, which allows _it_ to better reason around and assist the programmer, so in the spirit of meeting the machine halfway...

I will note my initial inclination _was_ to move the simple "negative" ( `Some(0) | None` ) condition first, which might make more sense. Happy to hear what others think, though.

Reply:
hmm, I think I would prefer to exhaustively match by starting with `Some(0) | None => false, Some(n)`.

It might still make sense to use `Some(n @ 1..)` as a quick reminder to the programmer, but in general I don't have a strong opinion here